src/niquests/adapters.py

```diff
@@ -1232,7 +1232,42 @@ def gather(self, *responses: Response, max_fetch: int | None = None) -> None:
                             self._orphaned.remove(low_resp)
 
                     if low_resp is None:
-                        low_resp = self.poolmanager.get_response()
+                        try:
+                            low_resp = self.poolmanager.get_response()
+                        except (ProtocolError, OSError) as err:
+                            raise ConnectionError(err)
+
+                        except MaxRetryError as e:
+                            if isinstance(e.reason, ConnectTimeoutError):
+                                # TODO: Remove this in 3.0.0: see #2811
```

Comment:
this seems to be requests ticket, rather niquests:
- https://github.com/psf/requests/issues/2811

perhaps use URL for these refs?